src/web/components/App.tsx

```diff
@@ -629,24 +637,181 @@ export const App = ({ CAPI, NAV }: Props) => {
-							title={documentBlockElement.title}
+							embedUrl={element.embedUrl}
+							height={element.height}
+							width={element.width}
+							title={element.title}
+						/>
+					</ClickToView>
+				</HydrateOnce>
+			))}
+			{CAPI.embedBlockElements.map((element) => (
+				<HydrateOnce
+					root="embed-block-element"
+					index={element.embedIndex}
+				>
+					{element.safe ? (
+						<ClickToView
+							role={element.role}
+							isTracking={element.isThirdPartyTracking}
+							source={element.source}
+							sourceDomain={element.sourceDomain}
+							abTests={CAPI.config.abTests}
+						>
+							<EmbedBlockComponent
+								key={element.embedIndex}
+								html={element.html}
+								alt={element.alt}
+							/>
+						</ClickToView>
+					) : (
+						<ClickToView
+							role={element.role}
+							isTracking={element.isThirdPartyTracking}
+							source={element.source}
+							sourceDomain={element.sourceDomain}
+							abTests={CAPI.config.abTests}
+							onAccept={() =>
+								updateIframeHeight(
+									`iframe[name="unsafe-embed-${
+										element.embedIndex || 0
+									}"]`,
+								)
+							}
+						>
+							<UnsafeEmbedBlockComponent
+								key={element.embedIndex}
+								html={element.html}
+								alt={element.alt || ''}
+								index={element.embedIndex || 0}
+							/>
+						</ClickToView>
+					)}
+				</HydrateOnce>
+			))}
+			{CAPI.instagramBlockElements.map((element) => (
+				<HydrateOnce
+					root="instagram-block-element"
+					index={element.instagramIndex}
+				>
+					<ClickToView
+						role={element.role}
+						isTracking={element.isThirdPartyTracking}
+						source={element.source}
+						sourceDomain={element.sourceDomain}
+						abTests={CAPI.config.abTests}
+					>
+						<InstagramBlockComponent
+							key={element.instagramIndex}
+							element={element}
+						/>
+					</ClickToView>
+				</HydrateOnce>
+			))}
+			{CAPI.mapBlockElements.map((element) => (
+				<HydrateOnce root="map-block-element" index={element.mapIndex}>
+					<ClickToView
+						role={element.role}
+						isTracking={element.isThirdPartyTracking}
+						source={element.source}
+						sourceDomain={element.sourceDomain}
+						abTests={CAPI.config.abTests}
+					>
+						<MapEmbedBlockComponent
+							format={format}
+							palette={palette}
+							embedUrl={element.embedUrl}
+							height={element.height}
+							width={element.width}
+							caption={element.caption}
+							credit={element.source}
+							title={element.title}
+						/>
+					</ClickToView>
+				</HydrateOnce>
+			))}
+			{CAPI.pullquoteBlockElements.map((element) => (
+				<HydrateOnce
+					root="pullquote-block-element"
+					index={element.pullquoteIndex}
+				>
+					<ClickToView
+						role={element.role}
+						isTracking={element.isThirdPartyTracking}
+						source={element.source}
+						sourceDomain={element.sourceDomain}
+						abTests={CAPI.config.abTests}
+					>
+						<PullQuoteBlockComponent
```

Comment:
As mentioned, this seems wild, but does this *only* hydrate when Composer *knows* there is an iframe?

Reply:
Yes so pullquotes will only be marked as 'tracking' if it contains an iframe with a src domain that isnt in our white list of non tracking domains. As such the overlay wont get rendered and it wont get hydrated.